src/import.ml

```diff
@@ -86,3 +92,27 @@ let with_confirmation message ~yes ?(no = "Cancel") f =
   match choice with
   | Some true -> f ()
   | _ -> Promise.return ()
+
+(** Builds application-specific functionality around [Vscode.Position] *)
+module Position = struct
+  include Vscode.Position
+
+  let t_of_jsonoo json =
+    let line = Jsonoo.Decode.(field "line" int) json in
+    let character = Jsonoo.Decode.(field "character" int) json in
+    Position.make ~line ~character
+
+  let t_to_jsonoo t = t |> Position.t_to_js |> Jsonoo.t_of_js
```

Comment:
`Jsonoo.t_of_js` isn't meant for encoding values.

You should use the `Jsonoo.Encode.*` functions.